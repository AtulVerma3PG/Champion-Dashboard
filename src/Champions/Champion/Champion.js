/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { DropdownButton, Dropdown } from "react-bootstrap";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSortUp, faSortDown, faSort } from "@fortawesome/free-solid-svg-icons";
import * as ReactBootStrap from "react-bootstrap";
import Pagination from "react-bootstrap/Pagination";
import SearchField from "react-search-field";
import withLayout from "../../hoc/withLayout";
import "../Css/Champion.css";
import "bootstrap/dist/css/bootstrap.min.css";

const championNumbers = [10, 20, 50];
/**
 * This Function is responsible to show the main champions Grid
 *
 * Main Champions Page
 * @param {object} props The details of the champions grid as props
 */
const Champion = (props) => {
  const watchlist = JSON.parse(localStorage.getItem("watchlist"));
  const [modWatchlist, setWatchlist] = useState(watchlist);
  const updateWatchlist = (watchlst) => {
    localStorage.setItem("watchlist", JSON.stringify(watchlst));
  };

  useEffect(() => {
    updateWatchlist(modWatchlist);
  });

  const {
    allChampions, pageSize, onSearchEnter, openWatchlist,
    searchedText, champions, firstPage, lastPage, setPageSize, sortBy, sortOn,
  } = props;

  /**
   *Remove Champion to Watchlist
   *
   * @param {integer} event ChampionId to add
   */
  const removeChampion = (event) => {
    const watch = modWatchlist.filter((e) => e.id !== event);
    setWatchlist(watch, () => { updateWatchlist(modWatchlist); });
  };

  /**
   *Add Champion to Watchlist
   *
   * @param {Array} event Champion to add
   */
  const addChampion = (event) => {
    setWatchlist((watchlsts) => [...watchlsts, event]);
    updateWatchlist(modWatchlist);
  };

  const champLength = allChampions.length;
  const expectedDivisns = champLength / pageSize;
  const divisions = champLength % pageSize === 0
    ? expectedDivisns
    : Math.floor(expectedDivisns) + 1;
  const items = [];
  for (let number = 1; number <= divisions; number += 1) {
    // Adding Pagination
    items.push(
      <Pagination.Item
        key={number}
        active={number === props.page}
        onClick={() => props.setPage(number)}
      >
        {number}
      </Pagination.Item>,
    );
  }

  return (
    <div>
      <div className="container">
        <div className="row">
          <div className="col left">
            <SearchField
              placeholder="Search item"
              className="styledButton"
              value={searchedText}
              onChange={onSearchEnter}
            />
          </div>
          <div className="col right">
            <div className="display-inlineb">
              {/* <p className="navText">Please select page size</p> */}
              <DropdownButton
                title={`PageSize ( ${pageSize} )`}
                id="document-type"
                className="display-inlineb"
                onSelect={(e) => setPageSize(e)}
              >
                {championNumbers.map((opt) => (
                  <Dropdown.Item as="button" eventKey={opt} key={opt}>
                    {opt}
                  </Dropdown.Item>
                ))}
              </DropdownButton>
            </div>
            <button
              type="button"
              className="styledButton"
              id="openWatchlist"
              onClick={openWatchlist}
            >
              Watchlist ({modWatchlist.length})
            </button>

          </div>
        </div>

      </div>
      <div className="container">
        <ReactBootStrap.Table responsive="md" bordered hover>
          <thead>
            <tr className="center">
              <th>Champion Profile</th>
              <th>
                <div className="heading">
                  <div>
                    Champion ID
                  </div>
                  <div className="sortIcon">
                    <i className="fa fa-sort-asc" style={sortBy === "Asc" && sortOn === "id" ? { color: "red" } : { color: "lightgray" }} onClick={() => props.sortAsc("id")} />
                    <i className="fa fa-sort-desc" style={sortBy === "Desc" && sortOn === "id" ? { color: "red" } : { color: "lightgray" }} onClick={() => props.sortDesc("id")} />
                    {/* <FontAwesomeIcon icon={faSortUp} style={sortBy === "Asc" && sortOn === "id" ? { color: "red" } : { color: "lightgray" }} size="1x" onClick={() => props.sortAsc("id")} />
                    <FontAwesomeIcon icon={faSortDown} style={sortBy === "Desc" && sortOn === "id" ? { color: "red" } : { color: "lightgray" }} size="1x" onClick={() => props.sortDesc("id")} /> */}
                  </div>
                </div>
              </th>
              <th>
                <div className="heading">
                  <div>
                    Name
                  </div>
                  <div className="sortIcon">
                    <FontAwesomeIcon icon={faSortUp} style={sortBy === "Asc" && sortOn === "name" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortAsc("name")} />
                    <FontAwesomeIcon icon={faSortDown} style={sortBy === "Desc" && sortOn === "name" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortDesc("name")} />
                  </div>
                </div>
              </th>
              <th>
                <div className="heading">
                  <div>
                    Armor
                  </div>
                  <div className="sortIcon">
                    <FontAwesomeIcon icon={faSortUp} style={sortBy === "Asc" && sortOn === "armor" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortAsc("armor")} />
                    <FontAwesomeIcon icon={faSortDown} style={sortBy === "Desc" && sortOn === "armor" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortDesc("armor")} />
                  </div>
                </div>
              </th>
              <th>
                <div className="heading">
                  <div>
                    armorperlevel
                  </div>
                  <div className="sortIcon">
                    <FontAwesomeIcon icon={faSortUp} style={sortBy === "Asc" && sortOn === "armorperlevel" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortAsc("armorperlevel")} />
                    <FontAwesomeIcon icon={faSortDown} style={sortBy === "Desc" && sortOn === "armorperlevel" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortDesc("armorperlevel")} />
                  </div>
                </div>
              </th>
              <th>
                <div className="heading">
                  <div>
                    hp
                  </div>
                  <div className="sortIcon">
                    <FontAwesomeIcon icon={faSortUp} style={sortBy === "Asc" && sortOn === "hp" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortAsc("hp")} />
                    <FontAwesomeIcon icon={faSortDown} style={sortBy === "Desc" && sortOn === "hp" ? { color: "#FF0000" } : { color: "lightgray" }} size="1x" onClick={() => props.sortDesc("hp")} />
                  </div>
                </div>
              </th>
              <th>Watchlist Action</th>
            </tr>
          </thead>
          <tbody>
            {champions.map((champion) => (
              <tr key={champion.id} className="center">
                <td>
                  <img src={champion.image_url} alt={champion.image_url} />
                </td>
                <td>{champion.id}</td>
                <td>
                  <Link
                    to={{
                      pathname: "/ChampionDetails",
                      state: { champion },
                    }}
                  >
                    {champion.name}
                  </Link>
                </td>
                <td>{champion.armor}</td>
                <td>{champion.armorperlevel}</td>
                <td>{champion.hp}</td>
                <td>
                  <div>
                    {modWatchlist.filter((e) => e.id === champion.id)
                      .length === 0 && (
                      <button
                        type="button"
                        className="btn btn-primary btn-lg"
                        onClick={() => addChampion(champion)}
                      >
                        Add
                      </button>
                    )}
                    {modWatchlist.filter((e) => e.id === champion.id)
                      .length === 1 && (
                      <button
                        type="button"
                        className="btn btn-primary btn-lg"
                        onClick={() => removeChampion(champion.id)}
                      >
                        Remove
                      </button>
                    )}
                  </div>
                </td>
              </tr>
            ))}
          </tbody>
        </ReactBootStrap.Table>
        {searchedText === "" && (
        <Pagination>
          <Pagination.First onClick={firstPage} />
          {items}
          <Pagination.Last onClick={lastPage} />
        </Pagination>
        )}
      </div>
    </div>
  );
};

Champion.propTypes = {
  allChampions: PropTypes.array.isRequired, // eslint-disable-line
  champions: PropTypes.array.isRequired, // eslint-disable-line
  setPageSize: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  pageSize: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  sortOn: PropTypes.string.isRequired,
  setPage: PropTypes.func.isRequired,
  watchlist: PropTypes.array.isRequired, // eslint-disable-line
  openWatchlist: PropTypes.func.isRequired,
  firstPage: PropTypes.func.isRequired,
  lastPage: PropTypes.func.isRequired,
  sortAsc: PropTypes.func.isRequired,
  sortDesc: PropTypes.func.isRequired,
  onSearchEnter: PropTypes.func.isRequired,
  searchedText: PropTypes.string.isRequired,
};

export default withLayout(Champion);
