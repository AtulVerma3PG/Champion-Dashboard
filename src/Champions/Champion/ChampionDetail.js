import React from "react";
import PropTypes from "prop-types";
import * as ReactBootStrap from "react-bootstrap";

const ChampionDetail = (props) => {
  const { location } = props;
  const { champion } = location.state;
  return (
    <div className="container">
      <h2>Champion Detail</h2>
      <br />
      <br />
      <br />

      <ReactBootStrap.Table responsive="lg" striped bordered hover>
        <thead>
          <th colSpan="2">
            <img src={champion.image_url} alt={champion.image_url} />
          </th>
        </thead>
        <tbody>
          <tr>
            <td>id </td>
            <td>
              {" "}
              {champion.id}
              {" "}
            </td>
          </tr>
          <tr>
            <td>name </td>
            <td>
              {" "}
              {champion.name}
              {" "}
            </td>
          </tr>
          <tr>
            <td>armor </td>
            <td>
              {" "}
              {champion.armor}
              {" "}
            </td>
          </tr>
          <tr>
            <td>armorperlevel </td>
            <td>
              {" "}
              {champion.armorperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackdamage </td>
            <td>
              {" "}
              {champion.attackdamage}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackdamageperlevel </td>
            <td>
              {" "}
              {champion.attackdamageperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackrange </td>
            <td>
              {" "}
              {champion.attackrange}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackspeedoffset </td>
            <td>
              {" "}
              {champion.attackspeedoffset}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackspeedperlevel </td>
            <td>
              {" "}
              {champion.attackspeedperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>crit </td>
            <td>
              {" "}
              {champion.crit}
              {" "}
            </td>
          </tr>
          <tr>
            <td>critperlevel </td>
            <td>
              {" "}
              {champion.critperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hp </td>
            <td>
              {" "}
              {champion.hp}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpperlevel </td>
            <td>
              {" "}
              {champion.hpperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpregen </td>
            <td>
              {" "}
              {champion.hpregen}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpregenperlevel </td>
            <td>
              {" "}
              {champion.hpregenperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>movespeed </td>
            <td>
              {" "}
              {champion.movespeed}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mp </td>
            <td>
              {" "}
              {champion.mp}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpperlevel </td>
            <td>
              {" "}
              {champion.mpperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpregen </td>
            <td>
              {" "}
              {champion.mpregen}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpregenperlevel </td>
            <td>
              {" "}
              {champion.mpregenperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>spellblock </td>
            <td>
              {" "}
              {champion.spellblock}
              {" "}
            </td>
          </tr>
          <tr>
            <td>spellblockperlevel </td>
            <td>
              {" "}
              {champion.spellblockperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>videogame_versions</td>
            <td>
              {champion.videogame_versions.map((version) => `${version} ,`)}
            </td>
          </tr>
        </tbody>
      </ReactBootStrap.Table>
    </div>
  );
};

ChampionDetail.propTypes = {
  location: PropTypes.object.isRequired,
};

export default ChampionDetail;
