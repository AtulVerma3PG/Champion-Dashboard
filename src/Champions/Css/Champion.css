button {
  text-align: center;
}
.searchBar {
  width: 20%;
}

.navText {
  color: aqua;
}
p {
  margin-left: 1rem;
  margin-right: 1rem;
}
.pageSize {
  margin-left: 1rem;
  margin-right: 1rem;
}

Typeahead {
  outline: none;
  position: relative;
}
.btn-arrow-left {
  position: relative;
  padding-left: 18px;
  padding-right: 18px;
}
.rbt {
  padding: 15px;
  margin: 10px 0 0;
  outline: none;
  position: relative;
}
Navbar {
  width: 100%;
}
.rbt .form-control {
  padding: 15px;
  margin: 5px 0 0;
  outline: none;
}

.rbt-token {
  width: fit-content;
}
