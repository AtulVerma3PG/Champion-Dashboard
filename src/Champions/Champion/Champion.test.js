/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import Champion from "./Champion";

const historyMock = { push: jest.fn() };
configure({ adapter: new Adapter() });
test("Champion Snapshot test", () => {
  const component = renderer.create(
    <Champion
      allChampions={[]}
      champions={[]}
      setPageSize={historyMock}
      page="0"
      pageSize="10"
      setPage={historyMock}
      addToWatchList={historyMock}
      removeFromWatchlist={historyMock}
      watchlist={[]}
      openWatchlist={historyMock}
      firstPage={historyMock}
      lastPage={historyMock}
      sort={historyMock}
      onSearchEnter={historyMock}
      searchedText=""
      openChampionDetails={historyMock}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});