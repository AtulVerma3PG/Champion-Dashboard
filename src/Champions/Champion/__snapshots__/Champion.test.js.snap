// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Champion Snapshot test 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      My Champions Dashboard
    </span>
    <div
      className="react-search-field "
      style={
        Object {
          "border": "1px #ddd solid",
          "display": "inline-flex",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <input
        className="react-search-field-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search item"
        style={
          Object {
            "border": "none",
            "color": "#5a5a5a",
            "flex": 1,
            "fontSize": 14,
            "fontWeight": 100,
            "height": 33,
            "outline": "none",
            "padding": 10,
          }
        }
        type="text"
        value=""
      />
      <button
        className="react-search-field-button"
        onClick={[Function]}
        style={
          Object {
            "appearance": "none",
            "backgroundColor": "white",
            "border": "none",
            "borderLeft": "1px #ddd solid",
            "boxSizing": "border-box",
            "cursor": "pointer",
            "height": 33,
            "outline": "none",
            "padding": 5,
            "width": 33,
          }
        }
        type="button"
      >
        <svg
          height={21}
          style={
            Object {
              "fill": "#727272",
            }
          }
          version="1.1"
          viewBox="0 0 635 635"
          width={21}
          x="0px"
          y="0px"
        >
          <g>
            <path
              d="M255.108,0C119.863,0,10.204,109.66,10.204,244.904c0,135.245,109.659,244.905,244.904,244.905 c52.006,0,100.238-16.223,139.883-43.854l185.205,185.176c1.671,1.672,4.379,1.672,5.964,0.115l34.892-34.891 c1.613-1.613,1.47-4.379-0.115-5.965L438.151,407.605c38.493-43.246,61.86-100.237,61.86-162.702 C500.012,109.66,390.353,0,255.108,0z M255.108,460.996c-119.34,0-216.092-96.752-216.092-216.092 c0-119.34,96.751-216.091,216.092-216.091s216.091,96.751,216.091,216.091C471.199,364.244,374.448,460.996,255.108,460.996z"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      className="display-inlineb"
    >
      <p
        className="navText"
      >
        Please select page size
      </p>
      <div
        className="pageSize display-inlineb dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-primary"
          disabled={false}
          id="document-type"
          onClick={[Function]}
          type="button"
        >
          10
        </button>
      </div>
    </div>
    <button
      className="btn btn-primary btn-lg"
      id="openWatchlist"
      onClick={
        Object {
          "push": [MockFunction],
        }
      }
      type="button"
    >
      Watchlist (
      0
      )
    </button>
  </nav>
  <div>
    <div
      className="table-responsive-md"
    >
      <table
        className="table table-bordered table-hover"
        stripped={true}
      >
        <thead>
          <tr>
            <th>
              Champion Profile
            </th>
            <th>
              <div
                className="heading"
              >
                <div>
                  Champion ID
                </div>
                <div
                  className="sortIcon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-up fa-w-10 fa-1x "
                    data-icon="sort-up"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-down fa-w-10 fa-1x "
                    data-icon="sort-down"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th>
              <div
                className="heading"
              >
                <div>
                  Name
                </div>
                <div
                  className="sortIcon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-up fa-w-10 fa-1x "
                    data-icon="sort-up"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-down fa-w-10 fa-1x "
                    data-icon="sort-down"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th>
              <div
                className="heading"
              >
                <div>
                  Armor
                </div>
                <div
                  className="sortIcon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-up fa-w-10 fa-1x "
                    data-icon="sort-up"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-down fa-w-10 fa-1x "
                    data-icon="sort-down"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th>
              <div
                className="heading"
              >
                <div>
                  armorperlevel
                </div>
                <div
                  className="sortIcon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-up fa-w-10 fa-1x "
                    data-icon="sort-up"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-down fa-w-10 fa-1x "
                    data-icon="sort-down"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th>
              <div
                className="heading"
              >
                <div>
                  hp
                </div>
                <div
                  className="sortIcon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-up fa-w-10 fa-1x "
                    data-icon="sort-up"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sort-down fa-w-10 fa-1x "
                    data-icon="sort-down"
                    data-prefix="fas"
                    focusable="false"
                    onClick={[Function]}
                    role="img"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </th>
            <th>
              Watchlist Action
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
          >
            «
          </span>
          <span
            className="sr-only"
          >
            First
          </span>
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
          >
            »
          </span>
          <span
            className="sr-only"
          >
            Last
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
